README.rst

```diff
@@ -0,0 +1,300 @@
+  The optimal significance is found to be at the cuts of `BDT > 0.1` and `PIDmu > 2`.
+
+.. code-block:: console
+
+   $ root -b -q Optimisation.C(<data_file>)
+
+
+Step 2. `ModelFixing.C <code/ModelFixing.C>`_ -
+  Creates the theoretical model that the fit is compared to.
+
+.. code-block:: console
+
+   $ root -b -q ModelFixing.C(<data_file>, <tmp_phi_models>)
+
+Step 3. `OSMassFit.C <code/OSMassFit.C>`_ -
+  The fit for the signal was modelled with the sum of Crystal Ball distributions. Each shape consists of a Gaussian core with a power law tail on opposite sides. The background was modelled with a 2nd order Chebyshev polynomial distribution.
+
+.. code-block:: console
+
+   $ root -b -q OSMassFit.C(<data_file>, <tmp_phi_models>, <result_plot>)
+
+
+
+3. Compute environment
+----------------------
+In order to be able to rerun the analysis even several years in the future, we
+need to "encapsulate the current compute environment", for example to freeze the
+ROOT version our analysis is using. We shall achieve this by preparing a `Docker
+<https://www.docker.com/>`_ container image for our analysis steps.
+
+Some of the analysis steps will run in a pure `ROOT <https://root.cern.ch/>`_
+analysis environment. We can use an already existing container image, for
+example `reana-env-root6 <https://github.com/reanahub/reana-env-root6>`_, for
+these steps.
+
+Some of the other analysis tasks wil need ``hftools`` Python library installed
+that our Python code needs. We can extend the ``reana-env-root6`` image to
+install ``hftools`` and to include our own Python code. This can be achieved as
+follows:
+
+.. code-block:: console
+
+   $ less environment/Dockerfile
+   # Start from the ROOT6 base image:
+   FROM reanahub/reana-env-root6
+
+   # Install HFtools:
+   RUN apt-get -y update && \
+       apt-get -y install \
+          python-pip \
+          zip && \
+       apt-get autoremove -y && \
+       apt-get clean -y
+   RUN pip install hftools
+
+   # Mount our code:
+   ADD code /code
+   WORKDIR /code
+
+We can build our analysis environment image and give it a name
+``reanahub/reana-demo-lhcb-d2pimumu``:
+
+.. code-block:: console
+
+   $ docker build -f environment/Dockerfile -t reanahub/reana-demo-lhcb-d2pimumu .
+
+We can push the image to the DockerHub image registry:
+
+.. code-block:: console
+
+   $ docker push reanahub/reana-demo-lhcb-d2pimumu
+
+(Note that typically you would use your own username such as ``johndoe`` in
+place of ``reanahub``.)
+
+
+
+4. Analysis workflow
+--------------------
+This analysis example consists of a simple workflow the theoretical model is generated and used for fitting.
+
+The analysis workflow consists of two above-mentioned stages:
+
+.. code-block:: console
+
+                START
+                 |
+                 | small.root
+                 |
+                 V
+   +-----------------------------+
```

Comment:
The steps (numbers and names) in the graph should correspond to steps mentioned earlier.

Reply:
Will actually update the workflow to include the resource consuming script `Optimise.C`